src/components/mal/AlternativGruppe.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'react';
+import { SituasjonAlternativ } from '../registreringsinfo/Alternativer';
+import { RadioPanelGruppe } from 'nav-frontend-skjema';
+import * as React from 'react';
+import { KnappeGruppe } from './Knappegruppe';
+
+interface AlternativContainerProps {
+    lagretSvar: string;
+    onSave: (valg: string) => void;
+    onCancel: () => void;
+}
+
+export function AlternativGruppe (props: AlternativContainerProps) {
+    const [svar, setSvar] = useState(props.lagretSvar);
+    const rdioAlternativer = Object.keys(SituasjonAlternativ)
+        .filter(x => SituasjonAlternativ[x] !== SituasjonAlternativ.IKKE_OPPGITT)
+        .map(alternativ => {
+                return {
+                    value: alternativ,
+                    label: SituasjonAlternativ[alternativ]
+                };
+            }
+        );
+
+    useEffect(() => {
+        setSvar(props.lagretSvar);
```

Comment:
Trenger du useEffect her ?
For meg så ser det ut som staten allerede inneholder det lagrede svaret 
`const [svar, setSvar] = useState(props.lagretSvar);  // lagretSvar er initalState`

Reply:
Ja, jeg skjønner hva du mener, må bruke useEffect her fordi endringer i `FremtidigSituasjon.situasjonState` skal trigge en endring av `props.lagretSvar i `AlternativGruppe`. Første renter så er `props.lagretSvar` tom fordi fetchkallet har ikke rukket å returnere noe enda, hvis jeg fjerner useEffect så vil ikke alternativet bli satt

Reply:
Hmm ? Er props.lagretSvar en funksjon ?
setSvar er jo bare en setState metode ? Denne trigger vel ikke noe ekstra ?

Reply:
nei, det er ikke en funksjon. Når det skjer en endring på en prop som en komponent er avhengig av så trigger det en re-render av komponenten. useEffect trigges hver gang det skjer en render hvis man kun sender ett parameter, men man kan også oppgi et parameter til for å filtrere hvilke endringer man vil at skal trigge useEffect. Så dvs at når FremtidigSituasjon komponentens fetch kall har oppdatert den lokale staten, og denne verdien sendes som prop til AlternativGruppe, så trigges render->useEffect i AlternativGruppe som setter markerer den korrekte checkboksen.

Reply:
err, tror du må vise meg på mandag :)